_posts/2020-05-03-dynamic-ipv6-playground.md

```diff
@@ -0,0 +1,121 @@
+[IPv6 configuration](https://tools.ietf.org/html/rfc4862).
+The data included in the RA message includes the network address
+(with its prefix length), routing information and potentially other extended
+network information (e.g. [DNS](https://tools.ietf.org/html/rfc8106) entries).
+
+### DHCPv6
+Although RA is sufficient for setting up a client host interface with IPv6
+configuration, DHCPv6 complements it with additional information in
+scenarios where additional options are in need.
+The latest [RFC](https://tools.ietf.org/html/rfc8415) even proposes a full
+alternative to the stateless address autoconfiguration (SLAAC), but is probably
+less common in the industry.
+
+DHCPv6 mainly complements SLAAC by allowing a specific IPv6 address to be set
+on an interface and the distribution of DNS information. It is also designed
+like its DHCPv4 origin with extendable options in mind, making it more
+accomodated for quick custom extensions.
+
+As the DNS information is already available in RA messages, DHCPv6 is mainly
+left suited for administrators that need to control the exact IPv6 full address
+(unlike SLAAC where only the network is distributed by RA and the host address
+is auto-generated).
+
+### dnsmasq
+We will use [dnsmasq](http://www.thekelleys.org.uk/dnsmasq/doc.html) to setup
+both an RA and DHCPv6 server to provide dynamic IPv6 services to the client.
+
+The server configuration is set on the *blue* namespace.
+
+- dnsmasq-config-ra-only
+```config
+leasefile-ro
+interface=veth10
+enable-ra
+dhcp-range=::,constructor:veth10,ra-only
+```
+
+- dnsmasq-config-slaac-dhcpv6
+```config
+leasefile-ro
+interface=veth10
+enable-ra
+dhcp-range=::2220,::2222,constructor:veth10,slaac
+```
+> **Note**: The DHCP server may also be configured to provide "static"
+addresses per specific identifiers (e.g. the interface mac address).
+This is useful for controlling the exact IPv6 address a requesting interface.
+With dnsmasq this translates to the following lines:
+```
+dhcp-host=12:34:56:78:90:ab,[fd00::22]
+dhcp-range=::,static
+```
+ 
+### Run Server
+- The ra-only server may be started as follow (in the blue namespace):
+```
+sudo ip netns exec blue dnsmasq \
+    --leasefile-ro \
+    --interface=veth10 \
+    --enable-ra \
+    --dhcp-range=::,constructor:veth10,ra-only
+```
+
+- To combine it with DHCPv6, run:
+```
+sudo ip netns exec blue dnsmasq \
+    --leasefile-ro \
+    --interface=veth10 \
+    --enable-ra \
+    --dhcp-range=::2220,::2222,constructor:veth10,slaac
+```
+
+> **Warning**: These operations may cause the existing DNS entries to get
+overwritten (i.e `resolv.conf`) by `dhclient`.
+
+## The Client
+
+On the client side (the red namespace) the IPv6 address should be already
+in place. This is the SLAAC one which its host bits have been auto-generated
+locally and the network bits delivered through the RA message.
+
+When DHCPv6 is also enabled, one can run a DHCPv6 client and the appropiate
```

Comment:
I am not sure I understand. Will it override the address we got in the previous step (line 105)?